Updated member list to show offline members also.

bbrockman opened the pull request:
Updated member list to show offline users and color the names according to their status.
In room.coffee: returned offline for the status in the method userActiveByUsername.  The method previously pulled the user from the session when doing the check.  It disregarded it if the user was not in the session (offline).
Commented out the base.less file that was overriding the status color in the member list.

bbrockman committed:
Added status-{{status}} to paragraph that displayed username in Member List.
In room.coffee: returned offline for the status in the method userActiveByUsername.  The method previously pulled the user from the session when doing the check.  It disregarded it if the user was not in the session (offline).
Commented out the base.less file that was overriding the status color in the member list.

rwakida commented:
Works for me :thumbsup: 

I think we should sort alphabetically.  Maybe group by status?

FYI: Duplicate of RocketChat/Rocket.Chat#337.  You should create a pull request for RocketChat.  Also, moderators suggest creating a configurable setting.

bbrockman commented:
Thanks Reid.  I need to go through their issues a little closer!

bbrockman committed:
Merge branch 'master' into feature/chat-locker-15

# Conflicts:
#	client/stylesheets/base.less
#	client/views/app/room.html

bbrockman committed:
Updated onlineUsers to allUsers for rooms because we want to see everyone in the room
Removed offset
Added profile first name and last name to user data publication
Updated userObserve to include new profile fields
Updated room to display first name and last name (sort by last name) instead of username

bbrockman committed:
Removed logging

bbrockman commented:
Added comment to userData publication to point out it is returning the data for all users in the publication and to not put any information that should be sent (profile.access).

bbrockman committed:
Added comment to not add additional fields to be returned with the publication because it now returns all users.

bbrockman committed:
Fixed Indenting error!  Thanks Coffee!

douglasrapp commented on client/lib/RoomManager.coffee:
```diff
@@ -105,16 +105,14 @@ Meteor.startup ->
 
 	updateUserStatus = (user, status, utcOffset) ->
-		onlineUsersValue = onlineUsers.curValue
+		allUsersValue = allUsers.curValue
```
Does **.curValue** return the same thing as **.get()** for a reactive var?

bbrockman replied:
I dunno.  I just changed their currrent one from online.  You're much more familiar with it.

douglasrapp commented on client/lib/RoomManager.coffee:
```diff
@@ -105,16 +105,14 @@ Meteor.startup ->
+			firstName: user.profile.first_name
+			lastName: user.profile.last_name
+			name: user.name
+			status: user.status
```
This isn't a bug here, but rather in how we're populating our users collection from LDAP. We're not populating a 'status' field on startup, so only users that have logged in before have it set.

The list of users on the right pane will use this field to determine how to style the name and the dot next to it (e.g., green == available, red == away). Normally, if a user is offline, it will appear as gray, making it obvious that the user is offline. However, in our case, if a user hasn't logged in before, the field is missing, so the name shows up in black with no dot.

I'll file an actual bug ticket about this. If we want, we can also work around it here by setting 'status' to:

```
status: if user.status? then user.status else 'offline'
```

but probably isn't worth it. Thoughts?

douglasrapp replied:
Ha ha! Looks like Reid made this a non-issue with pull request #50!

bbrockman replied:
Yep.  Fixed it while we were sleeping!

douglasrapp commented on client/stylesheets/base.less:
```diff
@@ -2605,9 +2605,8 @@ a.github-fork {
-			color: @secondary-font-color;
-			.calc(width,
-			~"100% - 45px");
+			//color: @secondary-font-color;
```
Do we want the whole name colored according to status, or just rely on the dot on the left of every name entry?

bbrockman replied:
I thought the whole name because the dot is so small.  And I like it better.

douglasrapp replied:
Works for me. It definitely makes it super clear/intuitive to tell when a user is offline.

douglasrapp commented on client/views/app/room.coffee:
```diff
@@ -149,7 +149,7 @@ Template.room.helpers
 
 	canPrivateMsg: ->
-		console.og 'room.helpers canPrivateMsg' if window.rocketDebug
+		console.log 'room.helpers canPrivateMsg' if window.rocketDebug
```
I prefer **console.og** better ;)

bbrockman replied:
I think someone used a url shortener while programming.

douglasrapp commented on client/views/app/room.coffee:
```diff
@@ -182,33 +182,25 @@ Template.room.helpers
-					username: username
-					status: onlineUsers[username]?.status
-					utcOffset: utcOffset
-
```
Nice. The UTC time offset was kinda cool, possibly useful-ish, but it definitely cluttered things up over there.

rwakida commented:
:+1:

bbrockman commented:
What do you know, now there are conflicts to merge it back in :disappointed:

javawebdeveloper committed:
Merge branch 'master' into feature/chat-locker-15

# Conflicts:
#	client/views/app/room.coffee
